test/e2e/tests/dapp-tx-edit.spec.js

```diff
@@ -0,0 +1,108 @@
+      },
+    ],
+  };
+  const smartContract = SMART_CONTRACTS.PIGGYBANK;
+  it('should NOT show an edit button on a contract interaction confirmation iniated by a dapp', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        smartContract,
+        title: this.test.title,
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // deploy contract
+        await driver.openNewPage(
+          `http://127.0.0.1:8080/?contract=${contractAddress}`,
+        );
+
+        // wait for deployed contract, calls and confirms a contract method where ETH is sent
+        await driver.findClickableElement('#deployButton');
+        await driver.clickElement('#depositButton');
+        await driver.waitUntilXWindowHandles(3);
+        const windowHandles = await driver.getAllWindowHandles();
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Deposit',
+        });
+        const editTransactionButton = await driver.isElementPresent(
+          '[data-testid="confirm-page-back-edit-button"]',
+        );
+        assert.equal(
+          editTransactionButton,
+          false,
+          `Edit transaction button should not be visible on a contract interaction created by a dapp`,
+        );
+      },
+    );
+  });
+
+  it('should show an edit button on a simple ETH send iniated by a dapp', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        smartContract,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage(`http://127.0.0.1:8080/`);
+
+        await driver.clickElement('#sendEIP1559Button');
+        await driver.waitUntilXWindowHandles(3);
+        const windowHandles = await driver.getAllWindowHandles();
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.waitForSelector({
+          css: '.confirm-page-container-summary__action__name',
+          text: 'Sending ETH',
```

Comment:
I think this is the problem:
```suggestion
          text: 'Deposit',
```

Reply:
Actually maybe not... still looking into this

Reply:
I don't think so, this is a simple send transaction. It shouldn't say "Deposit" any where?

Reply:
So there is a problem with the creation of the confirmation

Reply:
Yeah this is not it. It's the previous test that's failing, and it's failing because the edit button is present but invisible.